nginx_stage/lib/nginx_stage/views/pun_config_view.rb

```diff
@@ -126,7 +126,7 @@ def default_html_root
     # Array of env vars to declare in NGINX config using env directive
     # @return [Array<String>] list of env vars to declare in NGINX config
     def env_declarations
-      NginxStage.nginx_env(user: user).keys | scl_env_declarations | NginxStage.pun_custom_env_declarations
+      NginxStage.clean_nginx_env(user: user).keys | scl_env_declarations | NginxStage.pun_custom_env_declarations
```

Comment:
Maybe you add `scl_env_declarations`  and `NginxStage.pun_custom_env_declarations` into the `clean_nginx_env` and remove them from here to get rid of this statefulness we seem to have.

Reply:
To try and keep the code cleaner, added methods to define which env varrs to keep and merge that into ones we define and just pull all that into template.